cockatrice/src/player.cpp

```diff
@@ -300,6 +303,57 @@ Player::Player(const ServerInfo_User &info, int _id, bool _local, TabGame *_pare
     rfgMenu->addAction(aViewRfg);
     rfg->setMenu(rfgMenu, aViewRfg);
 
+    {
+        sharedMenu = playerMenu->addMenu(QString());
+
+        sharedMenu->addAction(tr("Draw card from shared"), [=]() {
```

Comment:
To be consistent with all the other actions I'd rather see these as separate functions than lambdas.

Reply:
The thing about that is, you have to capture in the action which player the action references.  In the long term I think you could use this approach to abstract out a  makeMenuForZone function as in #2657.  When you do it this way you also don't have to have a sperate retranslate step.  All in all I think switching everything to this you could cut things down 400ish lines.

That said this way of doing things doesnt exist in 5.5, so Ill switch it over to the old way when Im happy with the way it works.